### PR TITLE
Make taxon tree scroll horizontally on overflow

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

This fixes issue https://jira.czi.team/browse/IDSEQ-1005?filter=10801 by making the a horizontal scrollbar appear when the viewport width is less than 1200px.

# Notes

* I checked and there are no callers that set any of the affected viz options . 

* I tried a variety of approaches before settling on this very simple one, including 
  * shrinking to fit: made trees too small for many samples
  * determining size from container: didn't play nice with dynamic height resizing that depends on number of leaves

# Test

Make tree scroll:
image

Make tree not scroll:
image

### CODE REVIEW

**Comment by gregdingle** [comment]
@eggerdesigns welcome to review as well

**Comment by tfrcarvalho** [comment]
Do you know why these trees are short (only 3 levels). Would be good to see examples with all the levels.

File: app/assets/src/components/visualizations/TidyTree.js [hunk 1]
```diff
@@ -324,10 +326,15 @@ export default class TidyTree {
     }
 
     // adjust dimensions
-    let width = this.options.minWidth;
+    let width = Math.min(
+      Math.max(this.options.minWidth, this.options.width),
+      document.documentElement.clientWidth -
```
Comment (tfrcarvalho) [inline app/assets/src/components/visualizations/TidyTree.js]: We should adjust to the width of the parent container/div and not view port.
Specially because this is a generic tree. In other pages with elements left and right this might not work.

File: app/assets/src/components/visualizations/TidyTree.js [hunk 2]
```diff
@@ -14,8 +14,10 @@ export default class TidyTree {
         addOverlays: true,
         attribute: "aggregatescore",
         leafNodeHeight: 35,
-        minWidth: 1000,
-        minHeight: 300,
+        width: 1000,
```
Comment (tfrcarvalho) [inline app/assets/src/components/visualizations/TidyTree.js]: Do not set width and height directly. Just adjust to parent div, so users can set the size of the container as desired (css or js)

Comment (tfrcarvalho) [reply app/assets/src/components/visualizations/TidyTree.js]: Or at list make the defaults match the div below...

**Comment by tfrcarvalho** [comment]
Forgot to mention... to fully fix the issue you should add scrolling on the parent container (I believe heatmap does this).

**Comment by gregdingle** [comment]
> Do you know why these trees are short (only 3 levels)? Would be good to see examples with all the levels.

I don't know. My understanding is that the number of levels shown is determined dynamically. Are you suggesting to expand the tree? 

And I thought the mosquito project was a good example project. What example would recommend? 


> Do not set width and height directly. Just adjust to parent div, so users can set the size of the container as desired (css or js)

The width and height were already being set directly. I just changed the way the min and max were computed. Do you know why the size would not have been determined by the container originally? I guess it was something to do with SVG. 

> Dont think view port is the right reference to use. See comments.

To be clear, I added the viewport logic to cover a case outside of the motivating issue. It simply prevents the size from being set to a value that would always make the tree too big to be seen 100%. It's not clear that fitting the tree in some container is better than fitting the contain elements to the tree. I'd be happy to remove that logic if it is too confusing.

Related, I'm not side scrolling is better than fitting the tree to the content area, which is how it seems like it was designed to behave. I'd like to discuss that more.

**Comment by MarkAZhang** [comment]
Hey, I think there's a more elegant approach using `viewBox`, an avg attribute.

TLDR:
Instead of `<svg width={width} height={height}>`, do `<svg width="100%" viewBox="0 0 width height">`. (you'll need to do string interpolation, hard to do in a Github comment)

This will allow the visualization to scale with the width of the parent container, much like an image would scale. This works even if you resize the browser after the viz has loaded.
---

Basically, the `viewBox` specifies the coordinate system of all the elements inside the svg. If you do `viewBox="0 0 1 1"`, and then draw a point at x=0.5, y=0.5, then the point will be drawn in the center of the svg. (point is not a real svg element, this is just an example).

(if you don't specify a viewBox explicitly, the viewbox will default to `0 0 width height`

Once you have specified the viewBox, you can then specify a width and height for the svg element. The browser will then scale the svg to the width and height, while preserving the aspect ratio of the viewBox. So if I have `<svg viewBox="0 0 1 1" width="50" height="100">` then the browser will scale the svg to 50px by 50px (since the viewbox was 1 by 1), and center it.

File: app/assets/src/components/views/TaxonTreeVis.jsx [hunk 3]
```diff
@@ -352,6 +352,7 @@ class TaxonTreeVis extends React.Component {
           ref={container => {
             this.treeContainer = container;
           }}
+          style={{ overflowX: "scroll" }}
```
Comment (gregdingle) [inline app/assets/src/components/views/TaxonTreeVis.jsx]: I added this inline because there was no associated CSS file that I could find, and it seems likely this is will be a single instance

File: app/assets/src/components/visualizations/TidyTree.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ export default class TidyTree {
         addOverlays: true,
         attribute: "aggregatescore",
         leafNodeHeight: 35,
-        minWidth: 1000,
+        minWidth: 960,
```
Comment (gregdingle) [inline app/assets/src/components/visualizations/TidyTree.js]: This was necessary to prevent default scrolling on first load because the container maxes out at 1200px

File: app/assets/src/components/visualizations/TidyTree.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ export default class TidyTree {
         largerFont: 12,
         onCollapsedStateChange: () => {},
         collapsed: new Set(),
-        svgBackgroundColor: "white"
+        svgBackgroundColor: "white",
```
Comment (gregdingle) [inline app/assets/src/components/visualizations/TidyTree.js]: ignore this and below... prettier

**Comment by gregdingle** [comment]
@MarkAZhang @tfrcarvalho I spent another hour trying to make this work more elegantly but I didn't manage to do so. I think the fix here is good enough to ship. I talked with @eggerdesigns and she is okay with it as well.

File: app/assets/src/components/views/TaxonTreeVis.jsx (hunk 3 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/TaxonTreeVis.jsx]: Search for `taxon_tree_vis.scss`. It is old code, so still loaded in `loader.scss`.

**Comment by tfrcarvalho** [comment]
> @MarkAZhang @tfrcarvalho I spent another hour trying to make this work more elegantly but I didn't manage to do so. See notes in above at top.
> 
> I think the fix here is good enough to ship. I talked with @eggerdesigns and she is okay with it as well.

@gregdingle Can you change the title to match the change?

File: app/assets/src/components/views/TaxonTreeVis.jsx (hunk 3 above)
Comment (gregdingle) [reply app/assets/src/components/views/TaxonTreeVis.jsx]: thanks!
